### PR TITLE
feat(Edit Branch View): Allow deleting branch paths

### PR DESCRIPTION
**Pull request by geoffreykwan**
## Changes
- In the edit branch view you can now delete branch paths

Note:
CreateBranchPathsComponent should probably also be changed to use a FormArray like EditBranchPathsComponent.

Things that still need to be handled properly
- Deleting all paths
- Deleting all but one path
- Changing "How to Choose Among Paths", "Step", and "Component" has some bugs that still need to be worked on

## Test
- Make sure these actions work properly
  - Delete one path and submit, it should move the path steps after the branch
  - Delete multiple paths and submit, it should move the path steps after the branch
  - Add a path and delete an existing path and submit, it should handle this properly

### CODE REVIEW

File: src/assets/wise5/authoringTool/edit-branch-paths/edit-branch-paths.component.ts [hunk 1]
```diff
@@ -31,9 +37,67 @@ import { CreateBranchPathsComponent } from '../create-branch-paths/create-branch
   templateUrl: './edit-branch-paths.component.html'
 })
 export class EditBranchPathsComponent extends CreateBranchPathsComponent {
+  pathsFormArray: FormArray = new FormArray([]);
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/edit-branch-paths/edit-branch-paths.component.ts]: private?
```suggestion
  private pathsFormArray: FormArray = new FormArray([]);
```

File: src/assets/wise5/services/deleteBranchService.ts [hunk 2]
```diff
@@ -44,6 +44,30 @@ export class DeleteBranchService {
...
+  }
+
+  deleteBranchPathAndPlaceAfter(
+    branchPaths: any[],
+    branch: any,
+    branchStepId: string,
+    nodeIdToPlaceAfter: string,
+    nodeIdAfterMergeStep: string
+  ): void {
+    let placeAfterNodeId = nodeIdToPlaceAfter;
```
Comment (hirokiterashima) [inline src/assets/wise5/services/deleteBranchService.ts]: Do we need this ```placeAfterNodeId``` variable? Why can't we use ```nodeIdToPlaceAfter```?

File: src/assets/wise5/authoringTool/edit-branch/edit-branch.component.ts [hunk 3]
```diff
@@ -271,4 +259,29 @@ export class EditBranchComponent extends AbstractBranchAuthoringComponent {
       this.node.transitionLogic.whenToChoosePath = 'studentDataChanged';
     }
   }
+
+  private removePaths(branchPaths: any[], params: any): void {
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/edit-branch/edit-branch.component.ts]: Type CreateBranchParams? Maybe we want to rename CreateBranchParams to EditBranchParams?
Do we need to pass in branchPaths, or can we get it via ```this.branchPaths```?
```suggestion
  private removePaths(params: CreateBranchParams): void {
```

File: src/assets/wise5/authoringTool/edit-branch/edit-branch.component.ts [hunk 4]
```diff
@@ -271,4 +259,29 @@ export class EditBranchComponent extends AbstractBranchAuthoringComponent {
...
+    const nodeIdAfterMergeStep = this.projectService.getNodeIdAfter(params.mergeStepId);
+    let nodeIdToPlaceAfter = params.mergeStepId;
+    branchPaths
+      .filter((path: any) => path.delete)
+      .forEach((path: any, index: number) => {
+        if (path.delete) {
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/edit-branch/edit-branch.component.ts]: Do we need this check? Won't the filter guarantee that ```path.delete``` is true?
```suggestion
```

**Review by hirokiterashima — CHANGES_REQUESTED** [review]
1. I found a bug:
   1. Create a jigsaw lesson with 2 groups
   2. Delete the first path and add a new path with Choice Chosen = "Edit this to be branch 1 choice"
   3. Preview unit, and choose the first choice. It does not enable the next step arrow 
4. ```AbstractBranchAuthoringComponet.getBranchParams()``` is only used when creating a new branch, in CreateBranchComponent. Should it be moved to ```CreateBranchComponent.getBranchParams()```?
5. See inline comments

**Comment by geoffreykwan** [comment]
Ok, I've made the changes. I couldn't reproduce the jigsaw bug.

**Review by hirokiterashima — APPROVED** [review]
LGTM

**Pull request merged**